test/performance/shmem_perf_suite/bw_common.h

```diff
@@ -781,3 +811,58 @@ void static inline uni_dir_bw_main(int argc, char *argv[], bw_style bwstyl) {
     if (ret != -1)
         bw_finalize();
 } /*main() */
+
+static inline int check_hostname_validation(perf_metrics_t my_info) {
+
+    int hostname_status = -1;
+    int hostname_size = (HOST_NAME_MAX % 4 == 0) ? HOST_NAME_MAX : HOST_NAME_MAX + (4 - HOST_NAME_MAX % 4);
+    int i,j;
+
+    static long pSync_collect[SHMEM_COLLECT_SYNC_SIZE];
+    for (i = 0; i < SHMEM_COLLECT_SYNC_SIZE; i++)
+        pSync_collect[i] = SHMEM_SYNC_VALUE;
+
+    char *hostname = (char *) shmem_malloc (hostname_size * sizeof(char));
+    char *dest = (char *) shmem_malloc (my_info.num_pes * hostname_size * sizeof(char));
+
+    hostname_status = gethostname(hostname, hostname_size);
+    if (hostname_status == -1) {
+        fprintf(stderr, "gethostname returned -1. Exiting.\n");
+        return -1;
```

Comment:
Better to use `RETURN_ERROR_STR` here.  These macros will add extra debugging output including source file/line and PE id.

Reply:
Also, it would be better to check `hostname_status != 0` and print the exit code: `RETURN_ERROR_MSG("gethostname failed (%d)\n", hostname_status);`

Reply:
Should we just use `fprintf` to report the error? Since `RETURN_ERROR_MSG` brings the dependency on `shmem_internal.h` and it also does not get all the internal parameters (e.g. `shmem_internal_my_pe`) automatically.

Reply:
Sorry, I forgot this is in the unit tests.  You're right, we don't want to pull in shmem_internal.h here.